pkg/lassie/lassie.go

```diff
@@ -0,0 +1,60 @@
+package lassie
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/filecoin-project/lassie/internal"
+	"github.com/filecoin-project/lassie/pkg/retriever"
+	"github.com/filecoin-project/lassie/pkg/types"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-graphsync/storeutil"
+	carblockstore "github.com/ipld/go-car/v2/blockstore"
+)
+
+type Lassie struct {
+	timeout time.Duration
+}
+
+func NewLassie(timeout time.Duration) *Lassie {
+	return &Lassie{
+		timeout: timeout,
+	}
+}
+
+func (l *Lassie) Fetch(ctx context.Context, rootCid cid.Cid, outfile *os.File) (*types.RetrievalStats, types.RetrievalID, error) {
```

Comment:
oh, you're defining a top-level `Lassie`, so I guess that explains why you're doing an `internal`. BUT you don't need `putCBBlockstore`, you should move that back to `cmd` cause that's the only purpose it serves

Reply:
agree

Reply:
I feel like I'm missing something with LinkSystem, because if we're not making a LinkSystem from a blockstore, then what do we pass Retrieve()? Can I use an empty LinkSystem?

Reply:
OH, I might have figured it out. Does this look better?

```golang
blockstore, err := carblockstore.OpenReadWriteFile(outfile, []cid.Cid{rootCid})
if err != nil {
    return nil, types.RetrievalID{}, err
}
linkSystem := storeutil.LinkSystemForBlockstore(blockstore)
```

Reply:
yep